text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,484 @@
+  additionalOrigins: [
+    cloudfront.Origin.fromLoadBalancerV2(myLoadBalancer, {
+      pathPattern: '/api/*',
+      allowedMethods: AllowedMethods.ALL,
+      forwardQueryString: true,
+    }),
+  ],
+});
+```
+
+You can specify an origin group for your CloudFront origin if, for example, you want to configure origin failover for scenarios when you need high
+availability. Use origin failover to designate a primary origin for CloudFront plus a second origin that CloudFront automatically switches to when the
+primary origin returns specific HTTP status code failure responses. An origin group can be created and specified as the primary (or additional) origin
+for the distribution.
+
+```ts
+const myOriginGroup = cloudfront.Origin.fromOriginGroup({
+  primaryOrigin: cloudfront.Origin.fromLoadBalancerV2(myLoadBalancer),
+  fallbackOrigin: cloudfront.Origin.fromBucket(myBucket),
+  fallbackStatusCodes: [500, 503],
+});
+new cloudfront.Distribution(this, 'myDist', { origin: myOriginGroup });
+```
+
+The above will create both origins and a single origin group with the load balancer origin falling back to the S3 bucket in case of 500 or 503 errors.
+
+## Lambda@Edge
+
+Lambda@Edge is an extension of AWS Lambda, a compute service that lets you execute functions that customize the content that CloudFront delivers. You
+can author Node.js or Python functions in the US East (N. Virginia) region, and then execute them in AWS locations globally that are closer to the
+viewer, without provisioning or managing servers. Lambda@Edge functions are associated with a specific behavior and event type. Lambda@Edge can be
+used rewrite URLs, alter responses based on headers or cookies, or authorize requests based on headers or authorization tokens.
+
+By default, Lambda@Edge functions are attached to the default behavior:
+
+```ts
+const myFunc = new lambda.Function(...);
+const myDist = new cloudfront.Distribution(...);
+myDist.addLambdaFunctionAssociation({
+  functionVersion: myFunc.currentVersion,
+  eventType: EventType.VIEWER_REQUEST,
+});
+```
+
+Lambda@Edge functions can also be associated with additional behaviors, either at behavior creation or after the fact, either by attaching
+directly to the behavior, or to the distribution and referencing the behavior.
+
+```ts
+// Assigning at behavior creation.
+myOrigin.addBehavior('/images/*.jpg', {
+  viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  defaultTtl: cdk.Duration.days(7),
+  functionAssociation: [{
+    functionVersion: myFunc.currentVersion,
+    eventType: EventType.VIEWER_REQUEST,
+  }]
+});
+
+// Assigning after creation.
+const myImagesBehavior = myOrigin.addBehavior('/images/*.jpg', ...);
+myImagesBehavior.addFunctionAssociation({
+  functionVersion: myFunc.currentVersion,
+  eventType: EventType.VIEWER_REQUEST,
+});
+
+myDist.addFunctionAssociation({
+  functionVersion: myFunc.currentVersion,
+  eventType: EventType.ORIGIN_REQUEST,
+  behavior: myImagesBehavior,
+});
+```
+
```

Comment:
Would be nice to see how a full example looks like, i.e a single origin distribution with a custom domain serving https and an edge lambda.

Reply:
This is already present. See "Use case #3" in the Comparison section.